### PR TITLE
guide: Micronaut Framework Scopes

### PR DESCRIPTION
**Pull request by sdelamo**
Generate the guide locally: 

```
./gradlew micronautScopeTypesBuild
open build/dist/micronaut-scope-types.html 
```

I have intentionally left out `@Bean`. I think it can be [confusing to users](https://github.com/micronaut-projects/micronaut-core/issues/8058).  Moreover, I would like to encourage users to use `@Prototype` explicitly instead of just `@Bean`. 

[Micronaut Scope Types.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9614175/Micronaut.Scope.Types.pdf)

### CODE REVIEW

**Review by graemerocher — APPROVED** [review]

File: guides/micronaut-scope-types/java/src/test/java/example/micronaut/PrototypeScopeTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    HttpClient httpClient; // <2>
+
+//end::imports[]
+/*
+//tag::testheader[]
+        @Test
+        void prototypeScopeIndicatesThatANewInstanceOfTheBeanIsCreatedEachTimeItIsInjected() {
+            String path = "/";
+//end::testheader[]
+*/
```
Comment (alvarosanchez) [inline guides/micronaut-scope-types/java/src/test/java/example/micronaut/PrototypeScopeTest.java]: This is rendered really ugly:

image

Perhaps indent back 1 level the commented-out code?

File: guides/micronaut-scope-types/java/src/test/java/example/micronaut/RefreshableScopeTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    @Inject
+    @Client("/")
+    HttpClient httpClient; // <3>
+
+    @Test
+    void refreshableScopeIsACustomScopeThatAllowsABeansStateToBeRefreshedViaTheRefreshEndpoint() {
+//end::imports[]        
+/*
+//tag::testheader[] 
+       String path = "/";
```
Comment (alvarosanchez) [inline guides/micronaut-scope-types/java/src/test/java/example/micronaut/RefreshableScopeTest.java]: Extra space to align indention.

```suggestion
        String path = "/";
```

File: guides/micronaut-scope-types/java/src/test/java/example/micronaut/RefreshableScopeTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+//end::testheader[]
+*/
+//tag::test[]      
+        String path = "/refreshable";
+        BlockingHttpClient client = httpClient.toBlocking();
+        Set<String> responses = new HashSet<>(executeRequest(client, path));
+        assertEquals(1, responses.size()); // <1>
```
Comment (alvarosanchez) [inline guides/micronaut-scope-types/java/src/test/java/example/micronaut/RefreshableScopeTest.java]: Callout numbers from here are wrong

File: guides/micronaut-scope-types/java/src/test/java/example/micronaut/RequestScopeTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        String path = "/";
+//end::testheader[]
+*/
+        String path = "/request";
+//tag::test[]        
+        BlockingHttpClient client = httpClient.toBlocking();
+        Set<String> responses = new HashSet<>(executeRequest(client, path));
+        assertEquals(1, responses.size()); // <3>
+        responses.addAll(executeRequest(client, path));
+        assertEquals(2, responses.size()); // <4>
```
Comment (alvarosanchez) [inline guides/micronaut-scope-types/java/src/test/java/example/micronaut/RequestScopeTest.java:42]: Callout missing

File: guides/micronaut-scope-types/micronaut-scope-types.adoc [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+Micronaut features an extensible bean scoping mechanism based on https://javax-inject.github.io/javax-inject/[JSR-330]. 
+
+
+== Scenario
+
+We use the following scenario to talk about the different types of scopes.
+
+The following `@Controller` injects two collaborators. 
+
+source:singleton/RobotController[tags=pkg|imports|controller|clazz]
```
Comment (alvarosanchez) [inline guides/micronaut-scope-types/micronaut-scope-types.adoc:18]: Callouts are missing

File: guides/micronaut-scope-types/micronaut-scope-types.adoc [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+Each collaborator has an injection endpoint for a bean of type `Robot`.
```
Comment (alvarosanchez) [inline guides/micronaut-scope-types/micronaut-scope-types.adoc]:
```suggestion
Each collaborator has an injection point for a bean of type `Robot`.
```

File: guides/micronaut-scope-types/micronaut-scope-types.adoc [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+<4> Both injection points are fulfilled with the new instance of `Robot`.
+
+
+== Refreshable
+
+**https://docs.micronaut.io/latest/guide/#refreshable[Refreshable scope] is a custom scope that allows a bean's state to be refreshed via the https://docs.micronaut.io/latest/guide/#refreshEndpoint[/refresh endpoint].**
+
+source:refreshable/Robot[]
+
+Your application needs the `management` dependency to enable the refresh endpoint.
```
Comment (alvarosanchez) [inline guides/micronaut-scope-types/micronaut-scope-types.adoc:103]: I would also mention that it is sensitive by default

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

File: guides/micronaut-scope-types/micronaut-scope-types.adoc (hunk 7 above)
Comment (sdelamo) [reply guides/micronaut-scope-types/micronaut-scope-types.adoc:103]: done https://github.com/micronaut-projects/micronaut-guides/pull/1085/commits/695b51cc0c703d873f8d0a660d1b351fb3465334

File: guides/micronaut-scope-types/micronaut-scope-types.adoc (hunk 5 above)
Comment (sdelamo) [reply guides/micronaut-scope-types/micronaut-scope-types.adoc:18]: done https://github.com/micronaut-projects/micronaut-guides/pull/1085/commits/167b6676367d6846a5fcbc705a37d744e333ff11

File: guides/micronaut-scope-types/java/src/test/java/example/micronaut/RequestScopeTest.java (hunk 4 above)
Comment (sdelamo) [reply guides/micronaut-scope-types/java/src/test/java/example/micronaut/RequestScopeTest.java:42]: done via https://github.com/micronaut-projects/micronaut-guides/pull/1085/commits/bce481921ba63eb14bf6133ac7a9cc5b199fc0a2

File: guides/micronaut-scope-types/java/src/test/java/example/micronaut/PrototypeScopeTest.java (hunk 1 above)
Comment (sdelamo) [reply guides/micronaut-scope-types/java/src/test/java/example/micronaut/PrototypeScopeTest.java]: resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1085/commits/0f1be2e9d2d1435884a8f2d5a952db5e696d4610

**Review by alvarosanchez — APPROVED** [review]

**Pull request merged**